plugins/inputs/powerdns/powerdns_test.go

```diff
@@ -71,7 +74,8 @@ func (s statServer) serverSocket(l net.Listener) {
 func TestPowerdnsGeneratesMetrics(t *testing.T) {
 	// We create a fake server to return test data
 	randomNumber := int64(5239846799706671610)
-	socket, err := net.Listen("unix", fmt.Sprintf("/tmp/pdns%d.controlsocket", randomNumber))
+	sockname := filepath.Join(os.TempDir(), fmt.Sprintf("pdns%d.controlsocket", randomNumber))
```

Comment:
If we can add `ioutils`:
```suggestion
    sockname := ioutils.TempFile(os.TempDir(), "pdns*.controlsocket")
```

Reply:
I think TempFile also won't work (same as above) with sockets, so forget about this comment. :-)

Reply:
`ioutil.TempFile(...)` (or `ioutils.TempFile(...)` which is just wrapper to the former) creates file in filesystem - it is not recommended in this case: test will fail with `Cannot initialize socket` message (`net.Listen` is responsible for creating file and this file cannot exist in FS)

Reply:
Understood. Resolved.